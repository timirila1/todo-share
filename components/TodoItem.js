import React from 'react';
import { TouchableHighlight, Text, StyleSheet, View } from 'react-native';

export const TodoItem = (props) => {
    const { todo, navigation } = props;
    const onPress = () => {
        console.log('Yep');
    };

    return (
        <TouchableHighlight
            style={styles.underlay}
            underlayColor="white"
            onPress={onPress}
        >
            <View style={styles.item}>
                <Text style={styles.title}>{todo.title}</Text>
                <Text style={styles.desc}>{todo.desc}</Text>
            </View>
        </TouchableHighlight>
    );
};

const styles = StyleSheet.create({
    underlay: {
        paddingTop: 20,
        paddingLeft: 40,
        paddingRight: 20,
    },
    item: {
        flex: 1,
        borderBottomWidth: 1,
        borderBottomColor: 'rgba(0, 0, 0, 0.1)',
        paddingBottom: 10,
    },
    title: {
        fontSize: 24,
        fontWeight: '300',
        lineHeight: 36,
    },
    desc: {
        fontWeight: '200',
        color: 'rgba(0, 0, 0, 0.6)'
    }
});
