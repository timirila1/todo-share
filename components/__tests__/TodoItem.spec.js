import 'react-native';
import React from 'react';
import { TodoItem } from '../TodoItem';
import renderer from 'react-test-renderer';

it('Render TodoItem', async () => {
    const todoMock = {
        id: 1,
        title: 'Test title',
        desc: 'Test desc'
    };

    const tree = renderer.create(
        <TodoItem
            index={1}
            navigation={{}}
            todo={todoMock}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
