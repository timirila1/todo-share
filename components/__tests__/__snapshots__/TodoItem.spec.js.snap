// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TodoItem 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "paddingLeft": 40,
      "paddingRight": 20,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#e9e9e9",
        "borderBottomWidth": 1,
        "flex": 1,
        "paddingBottom": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "lineHeight": 36,
        }
      }
    >
      Test title
    </Text>
    <Text
      style={
        Object {
          "color": "#cacaca",
          "fontWeight": "200",
        }
      }
    >
      Test desc
    </Text>
  </View>
</View>
`;
